docs/sdg/sdg-flow-yaml.md

```diff
@@ -0,0 +1,164 @@
+pipeline = Pipeline.from_flows(ctx, ['mycustomflow.yaml'])
+```
+
+The YAML file format will mirror the API and look like this:
+
+```yaml
+version: 1.0
+blocks:
+  - name: gen_knowledge
+    type: LLMBlock
+    config: # LLMBlock constructor kwargs
+      output_cols: ["question", "response"]
+    gen_kwargs: # kwargs for block.generate()
+      max_tokens": 2048,
+    drop_duplicates: ["question"]
+  - name: filter_faithfulness
+    type: FilterByValueBlock
+    config:
+      filter_column: judgment
+      filter_value: YES
+      operation: eq
+      drop_columns: ["judgment", "explanation"]
+```
+
+## Versioning
+
+A mandatory `version` field in the YAML file expresses major and minor versions (e.g., 1.0, 1.1, 2.0).
+
+Compatibility rules
+
+1. If the major version of the YAML file is higher than the parser can handle, the parser should reject the file.
+2. If the minor version of the YAML file is higher than the highest version the parser is aware of, the parser should read the file but ignore any unrecognized content.
+3. If the file’s version is lower than the parser version, the parser should provide default values for any configuration introduced in later versions.
+
+Example parsing logic:
+
+```python
+def parse_custom_flow(content):
+    version = content['version']
+    major, minor = map(int, version.split('.'))
+
+    if major > PARSER_MAJOR:
+        raise IncompatibleVersionError("The custom flow file format is from a future major version.")
+    elif major <= PARSER_MAJOR and minor > PARSER_MINOR:
+        logger.warning("The custom flow file may have new features that will be ignored.")
+```
+
+### Flow Params
+
+The following parameters are supplied to a Flow constructor and used by the render() method:
+
+- client - an OpenAI completions API client for talking to the teacher model via the serving backend (i.e. llama-cpp or vLLM)
+- model_family - e.g. mixtral or merlinite
+- model_id - a path name for the specific teacher model being used
+- num_instructions_to_generate - how many samples to generate
+- batched - whether the model serving backend supports batched mode
+
+For now, we assume that the `Block` classes know how to use these parameters and there is no need to use any sort of templating in the custom flows that could use these parameters.
+
+### Model Prompts
+
+Based on whether model_family is mixtral or merlinite, a different prompt is used with the teacher model
+
+```python
+_MODEL_PROMPT_MIXTRAL = "<s> [INST] {prompt} [/INST]"
+_MODEL_PROMPT_MERLINITE = "'<|system|>\nYou are an AI language model developed by IBM Research. You are a cautious assistant. You carefully follow instructions. You are helpful and harmless and you follow ethical guidelines and promote positive behavior.\n<|user|>\n{prompt}\n<|assistant|>\n'"
+```
+
+For now, we assume that the `LLMBlock` class will choose the appropriate model prompt based on the family and that there is no need to specify a custom prompt.
+
+### Prompt Config Files
```

Comment:
why files over strings as in the YAML itself?
(i'm not saying the latter is better, just wondering how the decision is made here)

Reply:
No strong opinion from me on this - it just carried over from the current `Flow` design

But I could speculate - the content of these files would overwhelm the pipeline configuration and make the pipelines hard to reason about, and perhaps there's a use case for re-using the same prompt configs in different pipelines?

Reply:
Are we talking past each other? I'm referring to this: https://github.com/instructlab/sdg/blob/7bf156360f2c44b4c03570c1a67e0d63d6b69d53/src/instructlab/sdg/default_flows.py#L132-L134

Reply:
oh, you mean those block templates. i thought you refers to prompt template as defined per model family.

resolving as it's  my misunderstanding